unblob/handlers/filesystem/romfs.py

```diff
@@ -250,92 +265,52 @@ def create_symlink(self, extract_root: Path, output_path: Path, inode: FileHeade
         else:
             target_path = extract_root.joinpath(target).resolve()
 
-        if not is_safe_path(extract_root, target_path):
-            logger.warning(
-                "Path traversal attempt through symlink.", target_path=target_path
-            )
+        if not self.fs.check_path(target_path, "symlink"):
```

Comment:
Shouldn't this be handled by `self.fs.create_symlink(src=target_path, dst=output_path)` ?

Reply:
The problem with refactors/rewrites over many file, is that small details are glossed over. I have made this error many times, so I wanted to stay safe here and keep the original logic.

The code continues as
```py
        if not self.fs.check_path(target_path, "symlink"):
            return
        # we create relative paths to make the output directory portable
        output_path.symlink_to(os.path.relpath(target_path, start=output_path.parent))
```
It was left unchanged intentionally, because there was a path calculation *after* the check.
I wanted to keep the first check, then continue with creating the symlink.

However you are right, with your intuition, that the symlink should be created via `self.fs.create_symlink()`, so it was changed.